### PR TITLE
⚠ Expose Manager.Cache/Client options, deprecate older opts

### PR DESCRIPTION
**Pull request by vincepri**

### CODE REVIEW

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -130,6 +164,8 @@ type Options struct {
 	// is "done" with an object, and would otherwise not requeue it, i.e., we
 	// recommend the `Reconcile` function return `reconcile.Result{RequeueAfter: t}`,
 	// instead of `reconcile.Result{}`.
+	//
+	// Deprecated: Use Cache.SyncPeriod instead.
```
Comment (sbueringer) [inline pkg/manager/manager.go:168]: Should we replicate the godoc of this field over at the Cache.SyncPeriod?

File: pkg/manager/manager.go [hunk 2]
```diff
@@ -488,16 +525,27 @@ func (o Options) AndFrom(loader config.ControllerManagerConfiguration) (Options,
...
 	}
 
-	if o.Port == 0 && newObj.Webhook.Port != nil {
-		o.Port = *newObj.Webhook.Port
+	webhookServer := &webhook.Server{}
+	if newObj.Webhook.Port != nil {
+		if o.Port == 0 {
+			o.Port = *newObj.Webhook.Port
+		}
+		webhookServer.Port = *newObj.Webhook.Port
```
Comment (sbueringer) [inline pkg/manager/manager.go:533]: I'm probably missing something. Shouldn't webhookServer.Port be o.Port if o.Port is != 0?

Comment (sbueringer) [reply pkg/manager/manager.go:533]: Hm or is the idea that .Webhook.Port has the higher priority? It's just a bit confusing as in all the other places above we only set anything on o Options if the field there is not set

Comment (vincepri) [reply pkg/manager/manager.go:533]: `webhookServer.Port` is set regardless because it's really only set `if options.WebhookServer == nil {`

Comment (vincepri) [reply pkg/manager/manager.go:533]: And yeah this inheritance logic is very confusing to say the least, hence the deprecation :D

**Pull request merged**